### PR TITLE
Re-commit iceberg statistics and table handle revert

### PR DESCRIPTION
**Pull request by ZacBlanco**
## Description

This PR adds a separate commit with the prerequisite changes that are required when #22661 was merged. It re-adds the two additional commits so that they don't depend on the histogram PR.

## Motivation and Context

Re-commit some changes that were reverted in #22661 

## Impact

N/A

## Test Plan

N/A

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by hantangwangd — APPROVED** [review]
LGTM!

**Comment by jaystarshot** [comment]
Is the root cause of the revert https://github.com/prestodb/presto/pull/22661 identified?

**Review by jaystarshot — APPROVED** [review]

**Comment by ZacBlanco** [comment]
@jaystarshot Yes, it's identified and described in #22664. I'm looking to add some additional tests to verify that change doesn't have other adverse affects when histograms are used.

File: presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java [hunk 1]
```diff
@@ -203,10 +205,15 @@ private Node rewriteShowStats(ShowStats node, Table table, Constraint<ColumnHand
...
                         .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
             }
-            TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, ImmutableList.copyOf(columnHandles.values()), constraint);
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+            List<ColumnHandle> nonHiddenColumns = ImmutableList.copyOf(tableMetadata.getColumns().stream().filter(column -> !column.isHidden())
+                    .map(ColumnMetadata::getName)
+                    .map(columnHandles::get)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()));
+            TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, nonHiddenColumns, constraint);
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java:214]: I see that this part is originally from https://github.com/prestodb/presto/pull/22327
Can you explain what this change is and how does it affect the non-iceberg tables?

Comment (ZacBlanco) [reply presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java:214]: This prevents hidden columns from being queried when a user executes `SHOW STATS`. The `HiveMetadata` already implicitly filters out hidden columns when returning from `getTableStatistics` without a constraint.

**Review by feilong-liu — APPROVED** [review]

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**